### ISSUE TITLE
🚀 [Feature] On A Post Show User ID Button In User Hover Menu

### ISSUE THREAD

**Issue by Tirokk** [Human-Connection/Human-Connection#2784]
## :rocket: Feature

As a user I like to see the user profile ID of a user commented on a post or somewhere else.
It should show up if I hover over the user slug of a post or a comment and click on a button named `Show all user IDs`.

1. After the click all the **authors** IDs of **the post** and of **all comments** in the list on the post page shall appear. They shall be visible until I refresh the page.
2. The same shall happen if I hover over the post author slug and click in the same button on the menu which appears there. (is decided with @DennisHack 15.01.2020)

Suggestion:

<img width="1813" alt="Bildschirmfoto 2020-01-15 um 11 28 15" src="https://user-images.githubusercontent.com/25344101/72439888-108e0e00-37a8-11ea-9ccb-a949d8a5ff40.png">

### User Problem

If some users change their slugs or names I as a user like to have the ability to see a unique identifier to identify a user safely.

### Implementation

See above. ☝🏼

### Design & Layout

See above. ☝🏼

### Validation

None

### Additional context

None

**Comment by Tirokk** [comment]
My dear @alina-beck !
How you would design this nicely?
Please contact me personally ;-)

**Comment by alina-beck** [comment]
This is not so easy!

First off, I don't like the idea of showing all `user IDs` on a page by clicking a button hidden in the hover menu of _one_ user. And then hiding them again by reloading the page – this is not at all transparent and I would strongly advise against it from a UX perspective.

Also, why would I be able to see all `user IDs` on the `Post page` but nowhere else? Maybe I want to see the IDs of all post authors on the `index page` or check the IDs of users following another user, on the `profile page`. Why is that less important? And how do we explain to the users where this feature exists and where it doesn't? (It's also more difficult to implement – we'd have to check for the route to display or hide the button accordingly.)

Another way to implement this would be to have a `show user IDs` button somewhere globally available. I'm not sure where, though. In the `Avatar Menu`, maybe? Or in the `footer`? Either way, it should have the same effect on _all_ pages. It could also be a `setting`, similar to allowing `Embeds` or not.

And there's another issue: displaying the `user IDs` next to (or close to) the user slug will definitely mess up the design – on any page but the `Post page` where there's usually enough space. Except on mobile, of course.

One way to solve this would be to _replace_ the `username` or `date-time` below the slug with the `user ID` when the setting is active – when it's a temporary change I think this solution is preferable to a messed up layout.

@Tirokk @DennisHack what do you think? Let's maybe have a chat about this feature before we start building anything? :wink:

**Comment by alina-beck** [comment]
I also think there's an issue open – or did we just talk about it in some meeting? – to add the `user ID` to the `avatar hover menu`, so when hovering over an avatar you could see the `ID` of this user in the popup.

**Comment by Tirokk** [comment]
Not stale!

**Comment by rbeer** [comment]
#3093 removed the hover menu, didn't it?

**Comment by Tirokk** [comment]
Yes, you are right @rbeer ! 👍🏼

We go for #3007 and I close this now.
